src/analysis/retail/evoker/preservation/modules/talents/TitansGift.tsx

```diff
@@ -0,0 +1,170 @@
+    );
+  }
+
+  //Count total casts
+  countBlossomCasts() {
+    this.totalBlossomsCasted += 1;
+  }
+
+  countEchoCasts() {
+    this.totalEchoesCasted += 1;
+  }
+
+  //Track blossom healing added
+  emeraldBlossomHeal(event: HealEvent) {
+    const BlossomCast = getBlossomCast(event);
+    if (BlossomCast && isCastFromBurst(BlossomCast)) {
+      this.buffedBlossoms += 1;
+      const BlossomHeals = getHealEvents(event);
+      for (const BlossomHeal of BlossomHeals) {
+        this.healingAddedToBlossoms += (BlossomHeal.amount + (BlossomHeal.absorbed ?? 0)) / 5;
+      }
+    }
+  }
+
+  //Track echo healing added
+  echoHeal(event: HealEvent | ApplyBuffEvent | RefreshBuffEvent) {
+    const EchoApplication = getEchoAplication(event);
+    if (EchoApplication && isCastFromBurst(EchoApplication)) {
+      this.buffedEchoes += 1;
+      if (event.type === EventType.Heal) {
+        this.healingAddedToEcho += (event.amount + (event.absorbed ?? 0)) / 5;
+      } else {
+        if (event.ability.name === TALENTS_EVOKER.DREAM_BREATH_TALENT.name) {
+          const DbHealing = getDreamBreathHealing(event);
+          for (const DbHeal of DbHealing) {
+            this.healingAddedToEcho += (DbHeal.amount + (DbHeal.absorbed ?? 0)) / 5;
+          }
+        } else if (event.ability.name === TALENTS_EVOKER.REVERSION_TALENT.name) {
+          const RevHealing = getReversionHealing(event);
+          for (const RevHeal of RevHealing) {
+            this.healingAddedToEcho += (RevHeal.amount + (RevHeal.absorbed ?? 0)) / 5;
+          }
+        }
+      }
+    }
+  }
+
+  statistic() {
+    const PercentBuffedBlossoms =
+      this.totalBlossomsCasted !== 0 ? (this.buffedBlossoms * 100) / this.totalBlossomsCasted : 0;
+    const PercentBuffedEchoes =
+      this.totalEchoesCasted !== 0 ? (this.buffedEchoes * 100) / this.totalEchoesCasted : 0;
+
+    return (
+      <Statistic
+        position={STATISTIC_ORDER.OPTIONAL(5)}
+        size="flexible"
+        category={STATISTIC_CATEGORY.TALENTS}
+      >
+        <div className="pad">
+          <label>
+            <SpellLink spell={TALENTS_EVOKER.TITANS_GIFT_TALENT} />
+          </label>
+          <div className="value">
+            <div>
+              <small>
+                <SpellLink spell={SPELLS.EMERALD_BLOSSOM} />
+              </small>
+              <br />
+              <ItemHealingDone amount={this.healingAddedToBlossoms} />
+              <br />
+              <small>
+                {this.buffedBlossoms} casts buffed ({Math.trunc(PercentBuffedBlossoms)}%)
+              </small>
+            </div>
+            <div>
+              <small>
+                <SpellLink spell={TALENTS_EVOKER.ECHO_TALENT} />
+              </small>
+              <br />
+              <ItemHealingDone amount={this.healingAddedToEcho} />
+              <br />
+              <small>
+                {this.buffedEchoes} casts buffed ({Math.trunc(PercentBuffedEchoes)}%)
+              </small>
```

Comment:
I think it would be better to move the casts buffed into a `<TooltipElement>` similar to some other modules